flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -316,8 +316,28 @@ class ExprLowering {
   fir::ExtendedValue genval(const Fortran::evaluate::BOZLiteralConstant &) {
     TODO();
   }
-  fir::ExtendedValue genval(const Fortran::evaluate::ProcedureDesignator &) {
-    TODO();
+  fir::ExtendedValue
+  genval(const Fortran::evaluate::ProcedureDesignator &proc) {
+    const auto *symbol = proc.GetSymbol();
+    // TODO: non restricted specific intrinsics, needs to get runtime symbol,
+    // or think of something else.
```

Comment:
Do intrinsic procedures end up here then? I suspect we'll want to handle them separately, unless the handling is identical (no inline implementation) modulo name mangling.

Reply:
Yes, the following reference to `acos` ends up here:
```
 intrinsic :: acos
 call foo(acos)
```
I need to add an entry point to the intrinsic helper to return a funcOp pointing for the runtime (or the outline implementation) and use it here.

Reply:
Got it. Thanks.